plugins/Referrers/Columns/Base.php

```diff
@@ -438,19 +438,38 @@ protected function detectReferrerCampaignFromLandingUrl()
     }
 
 
+    /**
+     * Check if campaign parameters were directly provided in tracking request.
+     * This might e.g. be the case when using image tracking
+     *
+     * @param Request $request
+     * @return bool
+     */
     protected function detectReferrerCampaignFromTrackerParams(Request $request)
```

Comment:
This issue is meant to fix a "smaller" regression. Including bigger refactorings in it will only make it harder to review it.
In addition the campaign handling is something that needs to be refactored in general. See e.g. #19951

Reply:
And to be honest: Compared to the rest of the code, that method is actually small and easy to understand 😄